### PR TITLE
have slimy-update-snap.sh update curtin in the snap too

### PR DESCRIPTION
**Pull request by mwhudson**
Also tweak scripts/update-part.py to work better when switching
repositories.

### CODE REVIEW

File: scripts/slimy-update-snap.sh [hunk 1]
```diff
@@ -52,9 +52,12 @@ do_mount $old old
 
 add_overlay old new
 
+rm -rf new/lib/python3.6/site-packages/curtin
```
Comment (dbungert) [inline scripts/slimy-update-snap.sh:55]: What's your thoughts on doing probert also?  Does it get weird with the compiled libs?
If it's not too hard that would be nice.

**Review by dbungert — APPROVED** [review]
While the changes look reasonable, slimy still fails for me on this branch with a crash on start of the UI, with both the pending and non-pending impish isos.  Any thoughts?

**Comment by mwhudson** [comment]
> While the changes look reasonable, slimy still fails for me on this branch with a crash on start of the UI, with both the pending and non-pending impish isos. Any thoughts?

Are you running on an ISO that only has a stable version of the snap in it? That won't work, you need an ISO that has a newer snap (edge works).

File: scripts/slimy-update-snap.sh (hunk 1 above)
Comment (mwhudson) [reply scripts/slimy-update-snap.sh:55]: Hm. Yes compiled libs would be weird, although I guess it ought to be possible to replace the python code, which is all we usually care about. Maybe one for the next branch? :)

**Comment by dbungert** [comment]
> > While the changes look reasonable, slimy still fails for me on this branch with a crash on start of the UI, with both the pending and non-pending impish isos. Any thoughts?
> 
> Are you running on an ISO that only has a stable version of the snap in it? That won't work, you need an ISO that has a newer snap (edge works).

Ah, ok.  Presumably I can do a full snap build, inject, use that as my base iso, slimy, inject, and that sounds like it should work.  Trying it out.

**Comment by dbungert** [comment]
Yes, the above plan is working.  I can now script my way to a slimy-friendly daily base iso.
Ship it!

**Pull request merged**